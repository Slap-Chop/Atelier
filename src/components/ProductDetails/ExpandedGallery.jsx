import React from "react";
import ReactDOM from 'react-dom';
import Xbutton from './Images/Xbutton.png';

class ExpandedGallery extends React.Component {
  constructor(props){
    super(props);
    const overlayStyle = {
      position: 'fixed', /* Stay in place */
      zIndex: 1, /* Sit on top */
      left: 0,
      top: 0,
      width: '100%', /* Full width */
      height: '100%', /* Full height */
      justifyContent: 'center',
      display: 'flex',
      backgroundColor: 'rgb(0,0,0)', /* Fallback color */
      backgroundColor: 'rgba(0,0,0,0.4)' /* Black w/ opacity */
    }
  }



  render() {
    if (!this.props.open) {
      return null;
    }
    return (
    <div className='modal' style ={{
      position: 'fixed',
      zIndex: 1,
      left: 0,
      top: 0,
      width: '100%',
      height: '100%',
      justifyContent: 'center',
      display: 'flex',
      backgroundColor: 'rgb(0,0,0)',
      backgroundColor: 'rgba(0,0,0,0.4)'
    }}>
      <div className='modal-content' style={{
        width: '90%',
        height: '90%',
        backgroundImage:`url(${this.props.photos[this.props.photoIndex]?.url})`,
        backgroundRepeat: 'no-repeat',
        backgroundPosition: 'center',
        backgroundSize: 'contain',
      }}>
        <div className='modal-header'>
        <img
        src={Xbutton}
        onClick={(e) => this.props.xClick && this.props.xClick(e)}
        style={{
          width: '50px',
          height: '50px',
          display: 'flex',
          textAlign: 'center',
          textAlignVertical: 'center',
          paddingLeft: '5px',
          }}/></div>
        <div className='modal-body'></div>

      </div>
    </div>
    )
  }
}


export default ExpandedGallery;