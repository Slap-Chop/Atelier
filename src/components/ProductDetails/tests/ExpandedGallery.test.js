import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import ExpandedGallery from '../ExpandedGallery.jsx';

const expandedGallery = shallow(<ExpandedGallery
open={true}
xClick={jest.fn()}
photos={[{thumbnail_url: 'https://images.unsplash.com/photo-1534011546717-40…hcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80', url: 'https://images.unsplash.com/photo-1534011546717-40…cHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2734&q=80'}]}
photoIndex={0}
thumbnails={[{thumbnail_url: "https://images.unsplash.com/photo-1534011546717-407bced4d25c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
url: "https://images.unsplash.com/photo-1534011546717-407bced4d25c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2734&q=80"}]}
offset={0}
LRClick={false}
handleLeft={jest.fn()}
handleRight={jest.fn()}
handleUp={jest.fn()}
handleDown={jest.fn()}
handleThumb={jest.fn()}
/>)

test('expandedGallery should render', () => {
  expect(toJson(expandedGallery)).toMatchSnapshot();
})

test('Expanded Gallery should zoom on image click and pan on mouse move', () => {
  expandedGallery.find('img.modalImage').simulate('click');
  expect(expandedGallery.instance().state.zoomed).toEqual(true);
  expandedGallery.find('img.modalImage').simulate('mouseMove')
})