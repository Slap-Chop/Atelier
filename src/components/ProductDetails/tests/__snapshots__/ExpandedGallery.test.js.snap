// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expandedGallery should render 1`] = `
<div
  className="modal"
>
  <div
    className="modalContent"
  >
    <div
      className="modalImageContainer"
    >
      <img
        className="modalImage"
        onClick={[Function]}
        src="https://images.unsplash.com/photo-1534011546717-40â€¦cHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2734&q=80"
      />
    </div>
    <div
      className="modalHeader"
    >
      <img
        className="XButton"
        onClick={[Function]}
        src=""
      />
      <div
        className="iconContainer"
      >
        <ExpandedIcon
          click={[MockFunction]}
          index={0}
          key="0"
          photo={
            Object {
              "thumbnail_url": "https://images.unsplash.com/photo-1534011546717-407bced4d25c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
              "url": "https://images.unsplash.com/photo-1534011546717-407bced4d25c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2734&q=80",
            }
          }
          selected={1}
        />
      </div>
    </div>
    <div
      className="modal-body"
    />
  </div>
</div>
`;
